src/modules/transaction/schemas.rs

```diff
@@ -0,0 +1,157 @@
+    pub account_id: types::AccountId,
+    #[validate(custom = "crate::errors::validate_account_id")]
+    pub contract_id: Option<types::AccountId>,
+}
+
+#[derive(
+    Validate, Debug, Clone, PartialEq, Eq, serde::Serialize, serde::Deserialize, Apiv2Schema,
+)]
+pub struct ReceiptsRequest {
+    #[validate(custom = "crate::errors::validate_crypto_hash")]
+    pub transaction_hash: types::TransactionHash,
+}
+
+#[derive(
+    Validate, Debug, Clone, PartialEq, Eq, serde::Serialize, serde::Deserialize, Apiv2Schema,
+)]
+pub struct ActionReceiptsRequest {
+    #[validate(custom = "crate::errors::validate_account_id")]
+    pub account_id: types::AccountId,
+    #[validate(custom = "crate::errors::validate_account_id")]
+    pub contract_id: Option<types::AccountId>,
+}
+
+// *** Responses ***
+#[derive(Debug, Clone, PartialEq, Eq, serde::Serialize, serde::Deserialize, Apiv2Schema)]
+pub struct TransactionResponse {
+    pub transaction: Transaction,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, serde::Serialize, serde::Deserialize, Apiv2Schema)]
+
+pub struct TransactionsResponse {
+    pub transactions: Vec<Transaction>,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, serde::Serialize, serde::Deserialize, Apiv2Schema)]
+
+pub struct ReceiptsResponse {
+    pub receipts: Vec<Receipt>,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, serde::Serialize, serde::Deserialize, Apiv2Schema)]
+pub struct ActionReceiptsResponse {
+    pub action_receipts: Vec<Receipt>,
+}
+
+// *** Types ***
+
+#[derive(Debug, Clone, PartialEq, Eq, serde::Serialize, serde::Deserialize, Apiv2Schema)]
+pub struct Transaction {
+    /// Transaction Hash of the the transction
+    pub transaction_hash: String,
+    /// An account on which behalf transaction is signed
+    pub signer_account_id: String,
+    /// An access key which was used to sign the original transaction
+    pub signer_public_key: String,
+    /// Receiver account for this transaction
+    pub receiver_account_id: String,
+    /// The hash of the block this transaction was included in
+    pub block_hash: String,
+    /// A list of actions to be applied
+    pub actions: Vec<Action>,
+    /// Timestamp when the transaction was finalized
+    pub timestamp: u128,
+    /// Transaction cost in Yocto Near
+    pub total_gas_cost: u128, 
+    /// Amount of Near transferred during this transaction
+    pub amount: u128,
+    /// Status of the Transaction. Finalized | Pending | Failed
+    pub status: String,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, serde::Serialize, serde::Deserialize, Apiv2Schema)]
+pub struct Action {
+    /// Transaction Hash
+    pub transaction_hash: String,
+    /// The index of the action in the transaction
+    pub index_in_transaction: String,
+    /// Type of action 
+    pub action_kind: ActionType,
+    /// Arguments for the action
+    pub args: String,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, serde::Serialize, serde::Deserialize, Apiv2Schema)]
+pub struct Receipt {
```

Comment:
Do we want to limit Receipts to only Action Receipts? Or show users a list of data receipts as well?

Reply:
Explorer only shows action receipts I believe

Reply:
Good point on giving the devs as much data as possible as we do not know what they might use this data for yet. 

While I was trying to omit some fields to make the interface and responses "simpler", this helps clear things up a bunch on how I should move forward.

Reply:
> omit some fields to make the interface and responses "simpler"

this is OK unless we know what data they do require, it's kinda challenging task